event-store/impl-fs/src/commonMain/kotlin/dev/eskt/store/impl/fs/config.kt

```diff
@@ -68,8 +71,32 @@ public class FileSystemConfigBuilder(
         }
     }
 
+    @OptIn(ExperimentalSerializationApi::class, InternalSerializationApi::class)
     @Suppress("UNCHECKED_CAST")
     public fun <I : Any> createDefaultIdSerializer(type: KClass<I>): Serializer<I, String> {
+        return try {
+            val serializer = type.serializer()
+            object : Serializer<I, String> {
+                val cbor = Cbor
+                override fun serialize(obj: I): String {
+                    return cbor.encodeToHexString(serializer, obj)
+                }
+
+                override fun deserialize(payload: String): I {
+                    return cbor.decodeFromHexString(serializer, payload)
+                }
+            }
+        } catch (e: kotlinx.serialization.SerializationException) {
+            when (type) {
+                String::class -> object : Serializer<I, String> {
+                    override fun serialize(obj: I): String = obj as String
+                    override fun deserialize(payload: String): I = payload as I
+                }
+                else -> throw IllegalStateException(
+                    "$type is not marked with @Serializable and cannot be serialized automatically, please register this type with an explicit id serializer",
+                )
+            }
+        }
         return object : Serializer<I, String> {
```

Comment:
This is not reachable anymore, it was (mostly) absorbed above with the exception of the message now being generalised for both serialization and deserialization so I think it can be deleted

Reply:
good catch, forgot to delete it